packages/connector-azuread/src/types.ts

```diff
@@ -0,0 +1,34 @@
+import { z } from 'zod';
+
+export const azureADConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+  cloudInstance: z.string(),
+  tennantId: z.string(),
+});
+
+export type AzureADConfig = z.infer<typeof azureADConfigGuard>;
+
+export const accessTokenResponseGuard = z.object({
+  accessToken: z.string(),
+  scopes: z.array(z.string()),
+  tokenType: z.string(),
+});
+
+export type AccessTokenResponse = z.infer<typeof accessTokenResponseGuard>;
+
+export const userInfoResponseGuard = z.object({
+  id: z.string(),
+  displayName: z.string(),
+  given_name: z.string().optional().nullable(),
+  surname: z.string().optional().nullable(),
+  userPrincipalName: z.string().optional(),
+  JobTitle: z.string().optional().nullable(),
+  mail: z.string(),
+  MobilePhone: z.string().optional().nullable(),
+  officeLocation: z.boolean().optional().nullable(),
+  preferredLanguage: z.string().optional().nullable(),
+  businessPhones: z.array(z.string()).optional().nullable(),
+});
```

Comment:
- i found the response are all in camelCase
- `.optional().nullable()` equals to `.nullish()`
- `zod` only has `.partial()` to accept `undefined`, not including `null`, we still have to append `.nullish()` for every key

```suggestion
export const userInfoResponseGuard = z.object({
  id: z.string(),
  displayName: z.string().nullish(),
  givenName: z.string().nullish(),
  surname: z.string().nullish(),
  userPrincipalName: z.string().nullish(),
  jobTitle: z.string().nullish(),
  mail: z.string().nullish(),
  mobilePhone: z.string().nullish(),
  officeLocation: z.boolean().nullish(),
  preferredLanguage: z.string().nullish(),
  businessPhones: z.array(z.string()).nullish(),
});
```